src/cargo/ops/cargo_package/vcs.rs

```diff
@@ -173,7 +174,9 @@ fn git(
     // - ignored (in case the user has an `include` directive that
     //   conflicts with .gitignore).
     let mut dirty_files = Vec::new();
-    collect_statuses(repo, &mut dirty_files)?;
+    let pathspec = relative_pathspec(repo, pkg.root());
+    collect_statuses(repo, &[pathspec.as_str()], &mut dirty_files)?;
```

Comment:
> https://github.com/rust-lang/cargo/pull/14962 was closed because we found more bugs in cargo package, and https://github.com/rust-lang/cargo/pull/14962 could potentially make them even harder to fix. Two of them have been fixed so this is good to ship IMO with its own good.

My concern was about doing this in one collection but instead we have separate passes for files inside and files outside.  That works.

Reply:
That does mean we could have duplicate checks for the files outside but those should be on a different order of magnitude so it shouldn't be as much of an issue